src/senaite/jsonapi/api.py

```diff
@@ -1451,6 +1454,72 @@ def get_registry_records_by_keyword(keyword=None):
     return found_registers
 
 
+def get_settings_by_keyword(keyword=None):
+    """Get the settings associated to the specified keyword or, if
+     keyword is None, get all the settings.
+
+    :param keyword: settings to be retrieved
+    :return: dictionary with the settings plus a key to identify from which
+    keyword where retrieved.
+    """
+    key_to_ischema = {
+        'mail': [cp.mail.IMailSchema],
+        'calendar': [cp.calendar.ICalendarSchema],
+        'ram': [cp.ram.IRAMCacheSchema],
+        'language': [cp.language.ILanguageSelectionSchema],
+        'editing': [cp.editing.IEditingSchema],
+        'usergroups': [cp.usergroups.IUserGroupsSettingsSchema,
+                       cp.usergroups.ISecuritySchema,],
+        'search': [cp.search.ISearchSchema],
+        'filter': [cp.filter.IFilterAttributesSchema,
+                   cp.filter.IFilterEditorSchema,
+                   cp.filter.IFilterSchema,
+                   cp.filter.IFilterTagsSchema],
+        'maintenance': [cp.maintenance.IMaintenanceSchema],
+        'markup': [cp.markup.IMarkupSchema,
+                   cp.markup.ITextMarkupSchema,
+                   cp.markup.IWikiMarkupSchema,],
+        'navigation': [cp.navigation.INavigationSchema],
+        'security': [cp.security.ISecuritySchema],
+        'site': [cp.site.ISiteSchema],
+        'skins': [cp.skins.ISkinsSchema],
+    }
+
+    settings = []
+    if keyword is None:
```

Comment:
@ramonski from here to the return of the method the code is a bit ugly, but I had a hard time in finding a prettier way of achieving it.

Reply:
Thanks @juangallostra for your extra efforts!

Your code is nice and it fits on one screen to grasp the entire functionality without scrolling or navigating to different places.
So it is, despite its complexity, easy to understand and to maintain.

This is how code must look like;)

The only thing we can think of in terms of maximizing performance is to put the mapping `key_to_ischema` on module level, e.g.

```python
CONTROLPANEL_INTERFACE_MAPPING = {
    'mail': [cp.mail.IMailSchema],
    'calendar': [cp.calendar.ICalendarSchema],
    'ram': [cp.ram.IRAMCacheSchema],
    'language': [cp.language.ILanguageSelectionSchema],
    'editing': [cp.editing.IEditingSchema],
    'usergroups': [cp.usergroups.IUserGroupsSettingsSchema,
                            cp.usergroups.ISecuritySchema,],
    'search': [cp.search.ISearchSchema],
    'filter': [cp.filter.IFilterAttributesSchema,
                 cp.filter.IFilterEditorSchema,
                 cp.filter.IFilterSchema,
                 cp.filter.IFilterTagsSchema],
    'maintenance': [cp.maintenance.IMaintenanceSchema],
    'markup': [cp.markup.IMarkupSchema,
                      cp.markup.ITextMarkupSchema,
                      cp.markup.IWikiMarkupSchema,],
    'navigation': [cp.navigation.INavigationSchema],
    'security': [cp.security.ISecuritySchema],
    'site': [cp.site.ISiteSchema],
     'skins': [cp.skins.ISkinsSchema],
}
```